onnx/tensor.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include "onnx/onnx_pb.h"
+
+namespace onnx {
+
+struct Tensor {
+private:
+  bool is_segment_;
```

Comment:
do we actually support segmented/chunked tensors for serialization?

and I'd argue that for in-memory purposes even if we did - we should stitch them back together